samples/web/content/apprtc/gcm_register.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/python2.4
+#
+# Copyright 2015 Google Inc. All Rights Reserved.
+
+"""AppRTC GCM ID Registration
+
+This module implements user ID registration and verification.
+"""
+
+import constants
+import datetime
+import json
+import logging
+import webapp2
+
+from google.appengine.ext import ndb
+
+def get_ancestor_key():
+  return ndb.Key('GCMRecord', 'global')
+
+PARAM_USER_ID = 'userId'
+PARAM_GCM_ID = 'gcmId'
+PARAM_CODE = 'code'
+PARAM_OLD_GCM_ID = 'oldGcmId'
+PARAM_NEW_GCM_ID = 'newGcmId'
+PARAM_USER_ID_LIST = 'userIdList'
+
+class BindPage(webapp2.RequestHandler):
+  def get_message_from_json(self):
+    try:
+      message = json.loads(self.request.body)
+      return message
+    except Exception as e:
+      logging.warning('JSON load error from BindPage: error=' + str(e)
+          + ', request=' + self.request.body)
+      return None
+
+  def handle_new(self):
+    msg = self.get_message_from_json()
+    if msg and msg[PARAM_USER_ID] and msg[PARAM_GCM_ID]:
```

Comment:
Should this (and the other lines like it) use .has_key? I'm new to python but this appears to cause an error if the object doesn't have the key. Maybe we should centralize the message parsing/parameter retrieving into a helper.